torch_np/_funcs_impl.py

```diff
@@ -909,6 +909,22 @@ def take_along_axis(arr: ArrayLike, indices: ArrayLike, axis):
     return torch.take_along_dim(arr, indices, axis)
 
 
+def put(
+    a: NDArray,
+    ind: ArrayLike,
+    v: ArrayLike,
+    mode: NotImplementedType = "raise",
+):
+    v = v.type(a.dtype)
+    numel_ratio = ind.numel() / v.numel()
+    if numel_ratio.is_integer():
+        sizes = [int(numel_ratio)]
+        sizes.extend([1 for _ in range(v.dim() - 1)])
+        v = v.repeat(*sizes)
```

Comment:
A minor: When you want to see if an int divides another you do `a % b == 0`.
Now, this code still does nothing when `ind` and `v` do not have the same number of elements. I think the correct implementation would go along the lines of:
```suggestion
    # Nb. If ind.numel > v.numel() and v.numel() doesn't divide ind.numel()
    #     we're going to enter the first if, create more elements than necessary,
    #     and then trim the unnecessary elements in the second if
    if ind.numel() > v.numel():
        # Repeat v
        # ceil(ind.numel() / v.numel())
        ratio = (ind.numel() + v.numel() - 1) // v.numel();
        v = v.repeat(v.shape + (ratio,))
    if ind.numel() < v.numel():
        # Trim the last elements
        if v.ndim > 1:
            v = v.flatten()
        v = v[:ind.numel()]
```
Note that this has not been tested. Also, I doubt my code works when dealing with 0D arrays.

Reply:
Ah I see, makes sense! Pushed a change to this effect, notably

* I broadcasted v as opposed to repeat, as there were edge cases where repeating diverged from NumPy behaviour where broadcasting did not.
* I didn't check `if v.ndim > 1` <sup>(I imagine we would of wanted `v.ndim != 1` instead)</sup> before flattening, as I assume its a fast no-op anyway for 1D arrays?